rust/src/graph/cycles.rs

```diff
@@ -0,0 +1,29 @@
+use crate::errors::GrimpResult;
+use crate::graph::pathfinding;
+use crate::graph::{ExtendWithDescendants, Graph, ModuleToken};
+use rustc_hash::{FxHashMap, FxHashSet};
+use tap::Conv;
+
+impl Graph {
+    pub fn find_shortest_cycle(
+        &self,
+        module: ModuleToken,
+        as_package: bool,
+    ) -> GrimpResult<Option<Vec<ModuleToken>>> {
+        if as_package {
+            pathfinding::find_shortest_cycle(
+                self,
+                &module.conv::<FxHashSet<_>>().with_descendants(self),
```

Comment:
I am positively surprised that just using descendants here make it works. So we just treat the "importer" as a set of modules and simulate the situation where each of the descendants is importing everything that was originally imported by the provided "module". Right?

Reply:
`with_descendants` just expands the set of modules to include all the descendant modules e.g. `{"colors"}` becomes `{"colors", "colors.red", "colors.blue", "colors.green"}`